csrc/transformer/ds_transformer_cuda.cpp

```diff
@@ -880,6 +888,22 @@ std::vector<torch::Tensor> ds_transformer_backward(int layer_id,
         layer->SetSeqLength(seq_len, bsz);
     }
 
+    auto options = torch::TensorOptions()
+                       .dtype(g_output.options().dtype())
+                       .layout(torch::kStrided)
+                       .device(torch::kCUDA)
+                       .requires_grad(true);
+
+    auto workspace = torch::empty({get_workspace_size<T>(bsz,
+                                                         seq_len,
+                                                         layer->GetHiddenSize(),
+                                                         layer->GetIntermediateSize(),
+                                                         layer->GetNumHeads(),
+                                                         layer->IsTrainingMode(),
+                                                         layer->GeluCheckpoint())},
+                                  options);
```

Comment:
can we use g_output.options() here instead of creating a new option?

Reply:
good point 👍